rllib/utils/tf_ops.py

```diff
@@ -146,7 +146,7 @@ def zero_logps_from_actions(actions: TensorStructType) -> TensorType:
     # `deterministic_actions` or `stochastic_actions`). In case
     # actions are just [B], zeros_like works just fine here, but if
     # actions are [B, ...], we have to reduce logp back to just [B].
-    if len(logp_.shape) > 1:
+    while len(logp_.shape) > 1:
```

Comment:
This is the actual core change here, allowing action spaces of Box(n, m, shape=(x, y, ...), int32) to be supported.
This and the reinstantiation of the supported_spaces tests, making sure all these combinations (observation + action-spaces) work well.